ts/components/wallet/PayWebViewModal.tsx

```diff
@@ -0,0 +1,177 @@
+import URLParse from "url-parse";
+import BaseScreenComponent from "../screens/BaseScreenComponent";
+import { emptyContextualHelp } from "../../utils/emptyContextualHelp";
+import { RefreshIndicator } from "../ui/RefreshIndicator";
+import { useHardwareBackButton } from "../../features/bonus/bonusVacanze/components/hooks/useHardwareBackButton";
+import { isTestEnv } from "../../utils/environment";
+
+type Props = {
+  // the uri to send the form data thought POST
+  postUri: string;
+  // data to include into the form to submit
+  formData: Record<string, string | number>;
+  /**
+   * the path name that means the end of the process
+   * ex: the process ends when this url will be load https://yourdomain.com/finish/path/name
+   * finishPathName should be "/finish/path/name"
+   */
+  finishPathName: string;
+  /**
+   * the name of the query param used to include the outcome code
+   * ex: https://yourdomain.com/finish/path/name?myoutcome=123
+   * outcomeQueryparamName should be "myoutcome"
+   */
+  outcomeQueryparamName: string;
+  onFinish: (outcomeCode: Option<string>) => void;
+  onGoBack: () => void;
+  modalHeaderTitle?: string;
+};
+
+const styles = StyleSheet.create({
+  refreshIndicatorContainer: {
+    position: "absolute",
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
+    justifyContent: "center",
+    alignItems: "center",
+    zIndex: 1000
+  }
+});
+
+// the ID used as form ID and to identify it to ensure the autosubit
+const formId = `io-form-id-${uuid()}`;
+// the javascript submit command
+const injectedJSPostForm: string = `<script type="application/javascript">document.getElementById("${formId}").submit();</script>`;
+
+// create an html form giving a form data and an uri
+const crateAutoPostForm = (
+  form: Record<string, unknown>,
+  uri: string
+): string =>
+  `<form action="${uri}" method="post" id="${formId}" enctype="multipart/form-data" style="display: none;">
+    ${_.toPairs(form)
+      .map(kv => `<input type="text" name="${kv[0]}" value="${kv[1]}">`)
+      .join("<br/>")}
+  </form>`;
+
+/**
+ * return a 2-tuple
+ * 0 element: boolean. true is the given {@param url} contains the given {@param finishPathName} path name
+ * 0 element: string | undefined. a string contains the value of {@param outcomeQueryparamName} in query string of {@param url}
+ * @param url
+ * @param finishPathName
+ * @param outcomeQueryparamName
+ */
+const getFinishAndOutcome = (
+  url: string,
+  finishPathName: string,
+  outcomeQueryparamName: string
+): [isFinish: boolean, outComeCode: string | undefined] => {
+  const urlParse = new URLParse(url, true);
+  // find the object entry name in case insensitive
+  const maybeEntry = _.toPairs(urlParse.query).find(
+    kv => kv[0].toLowerCase() === outcomeQueryparamName.toLowerCase()
+  );
+  const outcome = maybeEntry ? maybeEntry[1] : undefined;
+  // found exit path
+  if (urlParse.pathname.toLowerCase().includes(finishPathName.toLowerCase())) {
```

Comment:
Since that we can't know what happens inside the WebView maybe is better to move this check as first step.
What do you think?

Reply:
What kind of scenario do you think could happen?
btw this function always return the outcome (the second value of 2tuple) so it must be computed anyway

Consider that this function is called by `handleOnShouldStartLoadWithRequest` that say if we want to load the given url or not inside the webview. So until we don't sya yes/not the webview is waiting for the response